src/cljs/main/broadfcui/page/method_repo/method/exporter.cljs

```diff
@@ -0,0 +1,204 @@
+   [broadfcui.common.table :refer [Table]]
+   [broadfcui.components.buttons :as buttons]
+   [broadfcui.components.modals :as modals]
+   [broadfcui.components.split-pane :refer [SplitPane]]
+   [broadfcui.components.workspace-selector :refer [WorkspaceSelector]]
+   [broadfcui.endpoints :as endpoints]
+   [broadfcui.page.method-repo.method.common :as method-common]
+   [broadfcui.page.workspace.workspace-common :as ws-common]
+   [broadfcui.net :as net]
+   [broadfcui.utils :as utils]
+   ))
+
+
+(react/defc- Preview
+  {:render
+   (fn [{:keys [state]}]
+     (let [{:keys [config config-error]} @state]
+       (cond config-error (style/create-server-error-message config-error)
+             config [:div {:style {:padding "0.5rem 1rem" :background-color "white"
+                                   :maxHeight "-webkit-fill-available"}}
+                     (method-common/render-config-details config)]
+             :else [comps/Spinner {:text "Loading Configuration Details..."}])))
+   :component-did-mount
+   (fn [{:keys [props this]}]
+     (this :-load (:preview-config props)))
+   :component-will-receive-props
+   (fn [{:keys [props next-props this]}]
+     (when (not= (:preview-config props) (:preview-config next-props))
+       (this :-load (:preview-config next-props))))
+   :-load
+   (fn [{:keys [state]} config]
+     (swap! state dissoc :config :config-error)
+     (let [{:keys [namespace name snapshotId]} config]
+       (endpoints/call-ajax-orch
+        {:endpoint (endpoints/get-configuration namespace name snapshotId true)
+         :on-done (net/handle-ajax-response
+                   (fn [{:keys [success? parsed-response]}]
+                     (if success?
+                       (swap! state assoc :config parsed-response)
+                       (swap! state assoc :config-error (:message parsed-response)))))})))})
+
+
+(defn- config->id+snapshot [config]
+  (assoc (select-keys config [:namespace :name]) :snapshotId (int (:snapshotId config))))
+
+
+(react/defc MethodExporter
+  {:get-initial-state
+   (fn [{:keys [props]}]
+     {:preview-config (some-> (:initial-config props) config->id+snapshot)})
+   :render
+   (fn [{:keys [props state this]}]
+     (let [{:keys [method-name dismiss]} props
+           {:keys [configs configs-error selected-config banner]} @state]
+       [modals/OKCancelForm
+        {:header (str "Export " method-name " to Workspace")
+         :content
+         (react/create-element
+          [:div {}
+           (when banner
+             [comps/Blocker {:banner banner}])
+           (cond configs-error (style/create-server-error-message configs-error)
+                 selected-config (this :-render-page-2)
+                 configs (this :-render-page-1)
+                 :else [comps/Spinner {:text "Loading Method Configurations..."}])])
+         :button-bar (cond selected-config (this :-render-button-bar-2)
+                           configs (this :-render-button-bar-1))
```

Comment:
I don't like the genericness of `-render-page-1`, `-render-page-2`, `-render-button-bar-1`, `-render-button-bar-2` ... how about more semantic names reflecting that 1 is about config selection and 2 is about workspace selection?